### PR TITLE
removing last app target now throws an exception

### PR DESCRIPTION
**Pull request by yuanxue-okta**
this is a similar change to #253 

@mystiberry-okta 
@alifatolahi-okta 
@brianzuzga-okta

### CODE REVIEW

**Comment by johngronberg-okta** [comment]
tagging @jakubvul for review since Mysti is out this week

File: _source/_docs/api/resources/roles.md [hunk 1]
```diff
@@ -480,7 +480,7 @@ HTTP/1.1 204 No Content
 
 Removes an app target from an `APP_ADMIN` role assignment.
 
-> Removing the last app target changes the scope of the role assignment from only applying to specific app targets to applying to **all** app targets.
+> Don't remove the last app target from a role assignment, as this causes an exception.  If you need a role assignment that applies to all apps, the API consumer should delete the `APP_ADMIN` role assignment and recreate it.
```
Comment (alifatolahi-okta) [inline _source/_docs/api/resources/roles.md]: "If you need a role assignment that applies to all apps, the API consumer should delete" this is a bit confusing. I'd change to something along the line of "If a role assignment is required to apply to all apps, the API consumer should delete"

Comment (yuanxue-okta) [inline _source/_docs/api/resources/roles.md]: this is the exact same wording as user admin's group target removal

Comment (jakubvul) [inline _source/_docs/api/resources/roles.md]: @alifatolahi-okta Isn't the correct term here "target"? That is, "If a role assignment should target all apps".

Comment (alifatolahi-okta) [inline _source/_docs/api/resources/roles.md]: yeah that makes sense, i.e. "If a role assignment should target all apps, the API consumer should delete".

@yuanxue-okta I find it confusing as it says "if you ..." and then assert "API consumer should ..." it's not clear whether "you" and "API consumer" are the same or different people. I'd consult with someone who has more experience and is a better domain expert; then decide as they advise.

Comment (yuanxue-okta) [inline _source/_docs/api/resources/roles.md]: changed to "If a role assignment should target all apps", ill loop in @brianzuzga-okta  and get his opinion as well

Comment (jakubvul) [inline _source/_docs/api/resources/roles.md]: I would rewrite this as follows:

"Removing the last app target will cause an exception. If a role assignment should target all apps, then you should delete the `APP_ADMIN` role assignment and recreate it."

However, it is not obvious what the connection between these two sentences is. They only make sense if you know that previously the documentation mentioned removing the last app target as a way of creating a role assignment that targets all apps. So right now it looks like we have two unrelated pieces of information here. Which is fine, but we should treat them that way.

Also, how do you create an `APP_ADMIN` role that targets all apps?

Comment (yuanxue-okta) [inline _source/_docs/api/resources/roles.md]: changed to the suggested wording.

all new APP_ADMIN will by default targets all apps, you can create an APP_ADMIN through an api POST call to  /api/v1/users/:uid/roles with a body of 
```
{
      "type": "APP_ADMIN"
}
```

**Comment by yuanxue-okta** [comment]
@alifatolahi-okta @jakubvul @brianzuzga-okta 
do you guys have any other concerns with the wording?

File: _source/_docs/api/resources/roles.md [hunk 2]
```diff
@@ -480,7 +480,7 @@ HTTP/1.1 204 No Content
 
 Removes an app target from an `APP_ADMIN` role assignment.
 
-> Removing the last app target changes the scope of the role assignment from only applying to specific app targets to applying to **all** app targets.
+> Removing the last app target will cause an exception. If a role assignment should target all apps, then you should delete the `APP_ADMIN` role assignment and recreate it.
```
Comment (brianzuzga-okta) [inline _source/_docs/api/resources/roles.md]: You may want to mirror the language that we have for group scoping of the UserAdmin role.

> Don’t remove the last group target from a role assignment, as this causes an exception. If you need a role assignment that applies to all groups, the API consumer should delete the USER_ADMIN role assignment and recreate it.

Comment (jakubvul) [inline _source/_docs/api/resources/roles.md]: Looks good to me!

**Pull request merged**